### DISCUSSION TITLE
How to add a custom class to a specific column header <th> tag?

### DISCUSSION THREAD

**Discussion by richcoleman-alphatrust** [TanStack/table#4824] [Q&A] [answered]
I'm using Vue 3.

I'm trying to add a class prop to a specific column header.

```
columnHelper.accessor((row) => row.id, {
            id: 'id',
            header: () => 'Options',  //<---I was to pass the prop to the header here
            cell: (props) => h(DataTableOptions, { id: props.row.original.id }),
        }),
```

I'm trying to make the Options column fit the contents of column (see the images for the difference I'm looking for).

I could easily just add the class directly in the HTML but I would like to pass it in as a prop for that one column since I don't want that class in all the other columns.

```
<thead class="bg-gray-50">
    <tr v-for="headerGroup in props.table.getHeaderGroups()" :key="headerGroup.id">
        <th
            v-for="header in headerGroup.headers"
            :key="header.id"
            :colspan="header.colSpan"
            @click="header.column.getToggleSortingHandler()?.($event)"
            class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900 hover:cursor-pointer">
            <template v-if="!header.isPlaceholder">
                <FlexRender :render="header.column.columnDef.header" :props="header.getContext()" />
                {{ { asc: ' ↑', desc: ' ↓' }[header.column.getIsSorted() as string] }}
            </template>
        </th>
    </tr>
</thead>
```

Currently the column width is just automatically determined.
![image](https://user-images.githubusercontent.com/23238954/233711763-ca5c8e07-388c-49ff-a837-99ab3fc719a2.png)

I'd like it to fit to the contents like this:
![image](https://user-images.githubusercontent.com/23238954/233711893-d554ec48-5bab-4299-b09f-bf63b884b93d.png)

**Comment by talatkuyuk** [comment, marked as answer]
You can add a property for the **class name** into column `meta`.
```js
declare module "@tanstack/table-core" {
  interface ColumnMeta<TData extends RowData, TValue> {
    className?: string;
  }
}
```
Then, you can define a className for the column that you want:
```js
columnHelper.accessor("options", {
  cell: // your implementation,
  header: // your implementation,
  footer: // your implementation,
  meta: {
    className: "classname-for-the-options-column",
  },
}),
```
Then, you can get it through the `column.columnDef.meta?.className`.
```jsx
// I do not know vue, it is for react, but you can got the idea.
<th className={column.columnDef.meta?.className ?? ""}> // also for <td>
```
Hope it is useful.

**Reply by richcoleman-alphatrust** [reply]
That was it!! Thank you so much!!

**Reply by marioteik** [reply]
This is so clean. Thanks!
